### PR TITLE
Add Donk Co. microwave board to Combat Bakery Kit

### PR DESCRIPTION
**Pull request by themias**
## About the PR

This PR adds a Donk Co. microwave board and instructional note to the Combat Bakery Kit (#31179). Donk Co. microwaves are able to bake baguette swords and throwing croissants (and normal recipes.) 

## Why / Balance

The SS13 version had the ability to bake more weapons, so this brings it in line. It adds opportunities to rearm yourself after eating past weapons to hide them, as well as arming other syndies.

## Technical details

The main hurdle was how to allow only some people to cook a recipe. We couldn't just learn it from a book like SS13, since we didn't want to just add a component onto a player entity like that.

I considered an implant, but that was a bit messier, since the implant would have the component rather than the player.

The microwave built with this board has a RecipeProviderComponent, which will add its recipes to the available list when cooking. The two new recipes are marked as SecretRecipe, and do not appear in the normal list.

## Media

https://github.com/user-attachments/assets/cc035c61-34dd-4e4c-aeae-1e37fd6b7921

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: The Donk Co. microwave can cook baguette swords and throwing croissants.
- tweak: The Combat Bakery Kit now has a Donk Co. microwave board and costs 6 TC.

### CODE REVIEW

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 1]
```diff
@@ -694,6 +697,28 @@ public override void Update(float frameTime)
             }
         }
 
+        /// <summary>
+        /// This is a helper method that tries to get secret recipes that the microwave might be capable of.
+        /// Currently, we only check the microwave itself, but in the future, the user might be able to learn recipes.
+        /// </summary>
+        private List<FoodRecipePrototype> TryGetSecretRecipes(EntityUid uid)
```
Comment (themias) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:704]: I'm realizing this probably could be an event instead.

**Comment by Golinth** [comment]
Maybe make the board the special donk co branded ones that the nukies have, and give this functionality to it instead? Its just odd to me to have two different syndicate microwaves

**Comment by themias** [comment]
> Maybe make the board the special donk co branded ones that the nukies have, and give this functionality to it instead? Its just odd to me to have two different syndicate microwaves

The idea is for it to be sneaky and look like a regular microwave. Those Nukie ones are bright red and would stand out.

**Comment by SlamBamActionman** [comment]
I think this is an interesting way to make it possible to cook weapons, but I would def make the board itself look sus. That way players can confirm the microwave is tampered with if they disassemble it, and if they do they likely suspect you already.

How will players learn the new recipes? If there is no way to know them outside reading a wiki/code it's bad. Maybe the kit can include a piece of paper listing the recipes (with a cheeky little "burn after reading" at the bottom)?

**Comment by themias** [comment]
> I think this is an interesting way to make it possible to cook weapons, but I would def make the board itself look sus. That way players can confirm the microwave is tampered with if they disassemble it, and if they do they likely suspect you already.
> 
> How will players learn the new recipes? If there is no way to know them outside reading a wiki/code it's bad. Maybe the kit can include a piece of paper listing the recipes (with a cheeky little "burn after reading" at the bottom)?

These are both good suggestions. I'll update the PR with them.

**Comment by ElPidor** [comment]
I made it a little cooler, how's it chaaat
image
image
[Traitor reciept.txt](https://github.com/user-attachments/files/16713579/Traitor.reciept.txt) if you'd like to put it in

**Comment by Admiral-Obvious-001** [comment]
I think this would be hilarious, but probably would need to be retuned a fair deal considering how easy it is to get the required materials with little effort.

**Review by UbaserB — CHANGES_REQUESTED** [review]
After internal discussion, we've decided that the microwave board should be the red donk co version so that nukies can also use it.

EDIT: I misinterpreted something, sorry for that.

**Comment by Everturning** [comment]
> I made it a little cooler, how's it chaaat image image [Traitor reciept.txt](https://github.com/user-attachments/files/16713579/Traitor.reciept.txt) if you'd like to put it in _I can change some parts if it doesnt fit wizden art style_

This isn't spam yall

**Pull request merged**